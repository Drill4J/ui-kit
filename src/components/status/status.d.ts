interface Props {
  className?: string;
  children?: string;
  testContext?: string;
}
export declare const Status: {
  ({ className, children, testContext }: Props): JSX.Element;
  displayName: string;
  __docgenInfo: {
    description: string;
    displayName: string;
    props: {
      className: {
        defaultValue: null;
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
      testContext: {
        defaultValue: {
          value: string;
        };
        description: string;
        name: string;
        required: boolean;
        type: {
          name: string;
        };
      };
    };
  };
};
export {};
