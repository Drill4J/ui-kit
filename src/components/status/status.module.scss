@import '../../style-guide/index';

.status {
  display: flex;
  align-items: center;
  font-size: 12;
  line-height: 16px;

  &::before {
    content: '';
    display: inline-block;
    margin-right: map-get($gutter-widths, micro);
    width: 8px;
    height: 8px;
    border-radius: 4px;
    background-color: currentColor;
  }
}
