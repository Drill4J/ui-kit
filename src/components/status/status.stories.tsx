import { storiesOf } from '@storybook/react';

import { Status } from './status';

storiesOf('Status', module)
  .add('with simple and bold text', () => (
    <>
      <div style={{ color: '#00B602' }}>
        <Status>Active</Status>
      </div>
      <div style={{ color: '#687481' }}>
        <Status>Ignored</Status>
      </div>
      <div style={{ color: '#ee0000' }}>
        <Status />
      </div>
    </>
  ));
