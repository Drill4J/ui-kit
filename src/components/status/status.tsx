import { BEM } from '@redneckz/react-bem-helper';

import styles from './status.module.scss';

interface Props {
  className?: string;
  children?: string;
  testContext?: string;
}

const status = BEM(styles);

export const Status = status(({ className, children, testContext = 'data-test' }: Props) => (
  <div className={className} data-test={`status:${testContext}`}>{children}</div>
));
