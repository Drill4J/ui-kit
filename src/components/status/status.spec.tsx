import React from 'react';
import renderer from 'react-test-renderer';
import { Status } from './status';

describe('Status', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Status />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
