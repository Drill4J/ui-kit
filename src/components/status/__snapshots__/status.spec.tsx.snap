// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status should match snapshot with simple text 1`] = `
<div
  className="status"
  data-test="status:data-test"
>
  Online
</div>
`;
